### PR TITLE
feat: Expanded checks

### PR DESCRIPTION
**Pull request by dcramer**
This adds a number of new checks (outlined below) as well as a test suite covering most checks.

- Switch license to MIT and add attribution to bugbear
- Rename existing B003 to B102
- Add checks from Bugbear: B003-B009
- Add test suite (modeled off of bugbear)
- Add pre-commit configuration

New Checks:

**B003**: Assigning to ``os.environ`` doesn't clear the
environment.  Subprocesses are going to see outdated
variables, in disagreement with the current process.  Use
``os.environ.clear()`` or the ``env=``  argument to Popen.

**B004**: Using ``hasattr(x, '__call__')`` to test if ``x`` is callable
is unreliable.  If ``x`` implements custom ``__getattr__`` or its
``__call__`` is itself not callable, you might get misleading
results.  Use ``callable(x)`` for consistent results.

**B005**: Using ``.strip()`` with multi-character strings is misleading
the reader. It looks like stripping a substring. Move your
character set to a constant if this is deliberate. Use
``.replace()`` or regular expressions to remove string fragments.

**B006**: Do not use mutable data structures for argument defaults.  All
calls reuse one instance of that data structure, persisting changes
between them.

**B007**: Loop control variable not used within the loop body.  If this is
intended, start the name with an underscore.

**B008**: Do not perform calls in argument defaults.  The call is
performed only once at function definition time.  All calls to your
function will reuse the result of that definition-time call.  If this is
intended, assign the function call to a module-level variable and use
that variable as a default value.

**B009**: Do not call ``getattr(x, 'attr')``, instead use normal
property access: ``x.attr``. Missing a default to ``getattr`` will cause
an ``AttributeError`` to be raised for non-existent properties. There is
no additional safety in using ``getattr`` if you know the attribute name
ahead of time.

**B010**: Do not call ``setattr(x, 'attr', val)``, instead use normal
property access: ``x.attr = val``. There is no additional safety in
using ``setattr`` if you know the attribute name ahead of time.

### CODE REVIEW

**Comment by dcramer** [comment]
Sentry has a number of violations around this that we'll need to address - though we can simply exclude these new checks in setup.cfg until they pass.

**Comment by dcramer** [comment]
B007 is likely going to cause some complaints - we could axe it as it seems opinionated to me. That said there's certainly value in making it harder to reference those variables since they leak scope.

File: sentry_check.py [hunk 1]
```diff
@@ -183,149 +293,186 @@ def load_file(self):
...
+B010 = Error(
+    message=u"B010: Do not call setattr with a constant attribute value, "
+    "it is not any safer than normal property access."
 )
 
-B101 = partial(error, type=SentryCheck)
+B101 = Error(
+    message=u"B101: Avoid using the {} mock call as it is "
+    "confusing and prone to causing invalid test "
+    "behavior."
```
Comment (markstory) [inline sentry_check.py:383]: We might have a few of these in sentry.

File: sentry_check.py [hunk 2]
```diff
@@ -183,149 +293,186 @@ def load_file(self):
...
+    "collections.deque",
+    "defaultdict",
+    "deque",
+    "dict",
+    "list",
+    "set",
+}
+B007 = Error(
+    message=u"B007: Loop control variable {!r} not used within the loop body. "
+    "If this is intended, start the name with an underscore."
```
Comment (markstory) [inline sentry_check.py:361]: We'll have this one for sure, but I think this is a good rule.

File: sentry_check.py (hunk 1 above)
Comment (mattrobenolt) [reply sentry_check.py:383]: This isn't new, we already do this.

**Comment by dcramer** [comment]
Anyone have objections? We can wait to update Sentry til we exclude the ones thatll be problematic (the for loop one).

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**